sims/verilator/Makefile

```diff
@@ -1,112 +1,175 @@
 # general path variables
-#########################################################################################
+#============================================================================
 base_dir=$(abspath ../..)
 sim_dir=$(abspath .)
 
-#########################################################################################
+#============================================================================
 # include shared variables
-#########################################################################################
+#============================================================================
 include $(base_dir)/variables.mk
 
-#########################################################################################
+#============================================================================
 # name of simulator (used to generate *.f arguments file)
-#########################################################################################
+#============================================================================
 sim_name = verilator
 
-#########################################################################################
-# vcs simulator types and rules
-#########################################################################################
+#============================================================================
+# verilator simulator types and rules
+#============================================================================
 sim_prefix = simulator
 sim = $(sim_dir)/$(sim_prefix)-$(MODEL_PACKAGE)-$(CONFIG)
 sim_debug = $(sim_dir)/$(sim_prefix)-$(MODEL_PACKAGE)-$(CONFIG)-debug
 
-PERMISSIVE_ON=
-PERMISSIVE_OFF=
-
-WAVEFORM_FLAG=-v$(sim_out_name).vcd
-
 .PHONY: default debug
 default: $(sim)
 debug: $(sim_debug)
 
-#########################################################################################
+#============================================================================
 # import other necessary rules and variables
-#########################################################################################
+#============================================================================
 include $(base_dir)/common.mk
 
-#########################################################################################
-# verilator binary and flags
-#########################################################################################
+#============================================================================
+# selectively enabling printfs
+#============================================================================
+ENABLE_PRINTF_PATTERN ?=
+sim_vsrcs_printf := $(sim_vsrcs:.v=.printf.v)
+
+#============================================================================
+# verilator/cxx binary and flags
+#============================================================================
+# can't use -Wall, firrtl spits out 1000's of warnings that flag as erros
+#VERILATOR := verilator --cc --exe -Wall
```

Comment:
Probably should remove this instead of commenting

Reply:
i will remove line 64, but 63 is documentation. i would like to include -Wall but i can't

Reply:
Certainly, I agree